Source/WebCore/style/CustomPropertyRegistry.cpp

```diff
@@ -83,16 +83,8 @@ bool CustomPropertyRegistry::registerFromAPI(CSSRegisteredCustomProperty&& prope
 
 void CustomPropertyRegistry::registerFromStylesheet(const StyleRuleProperty::Descriptor& descriptor)
 {
-    // "The inherits descriptor is required for the @property rule to be valid; if it’s missing, the @property rule is invalid."
-    // https://drafts.css-houdini.org/css-properties-values-api/#inherits-descriptor
-    if (!descriptor.inherits)
-        return;
-
-    // "If the provided string is not a valid syntax string, the descriptor is invalid and must be ignored."
-    // https://drafts.css-houdini.org/css-properties-values-api/#the-syntax-descriptor
     auto syntax = CSSCustomPropertySyntax::parse(descriptor.syntax);
```

Comment:
I think StyleRuleProperty::Descriptor can now include syntax as CSSCustomPropertySyntax instead of a String. This way you won't have to parse again here.

Reply:
Indeed it's a bit annoying to double parse, but we need to keep the user-written syntax for serialization. https://drafts.css-houdini.org/css-properties-values-api/#serialize-a-csspropertyrule

Reply:
Indeed it's a bit annoying to double parse, but we need to keep the user-written syntax for serialization. https://drafts.css-houdini.org/css-properties-values-api/#serialize-a-csspropertyrule